core/Tracker/Visit.php

```diff
@@ -166,6 +166,12 @@ public function handle()
         // AND
         // - the last page view for this visitor was less than 30 minutes ago @see isLastActionInTheSameVisit()
         if (!$isNewVisit) {
+            // on a ping request that is received before the standard visit length, we just update the visit time w/o
+            // adding a new action
+            if ($this->isPingRequest()) {
+                $action = null;
+            }
+
```

Comment:
FYI, added an item about ignoring conversion stuff for ping in https://github.com/piwik/piwik/issues/8071